src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -382,16 +394,17 @@ TCPTransactionId RTCPMessageManager::sendLogicalPortIsClosedRequest(
     return sendLogicalPortIsClosedRequest(p_channel_resource, request);
 }
 
-TCPTransactionId RTCPMessageManager::sendUnbindConnectionRequest(TCPChannelResource *p_channel_resource)
+TCPTransactionId RTCPMessageManager::sendUnbindConnectionRequest(
+        std::shared_ptr<TCPChannelResource>& channel)
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.
Coding style (one line).